### PR TITLE
Add vinyl to cart

### PR DESCRIPTION
**Pull request by IonesiCristi**
Mainly I added the endpoint for adding a vinyl to cart,created all the "infrastructure" needed for this (models,repositories etc.), added the required validations for this and also added some more validations for create vinyl endpoint

### CODE REVIEW

File: src/main/java/com/fortech/controller/VinylController.java [hunk 1]
```diff
@@ -20,7 +22,7 @@
 public class VinylController {
 
 	@Autowired
-	private VinylService vinylService;
+	private VinylService vinylServiceImpl;
```
Comment (nicolaevieru) [inline src/main/java/com/fortech/controller/VinylController.java]: why Impl

File: src/main/java/com/fortech/model/Vinyl.java [hunk 2]
```diff
@@ -1,36 +1,50 @@
...
 	@Min(0)
-	int stock;
+	private int stock = 0;
+	
+	public Vinyl() {
+	}
 
-	public Long getId() {
+	public Vinyl(VinylCreateDto vinyl) {
+		super();
```
Comment (nicolaevieru) [inline src/main/java/com/fortech/model/Vinyl.java]: why call super?

Comment (IonesiCristi) [inline src/main/java/com/fortech/model/Vinyl.java]: It was auto-generated when I generated the constructor,it is redundant indeed.

File: src/main/java/com/fortech/controller/VinylController.java (hunk 1 above)
Comment (IonesiCristi) [inline src/main/java/com/fortech/controller/VinylController.java]: I refactored the name of a class and this field was changed to,I will modify it.

**Pull request merged**